### PR TITLE
PRC-333: Contact address phone sync endpoints, services, events and tests

### PR DESCRIPTION
**Pull request by timharrison-moj**

### CODE REVIEW

**Review by james-baker-moj — APPROVED** [review]
LGTM

**Review by prabash-moj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/sync/ContactAddressPhoneSyncController.kt [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    @Parameter(description = "The internal ID for the address-specific phone number", required = true)
+    @PathVariable contactAddressPhoneId: Long,
+  ) = syncFacade.deleteContactAddressPhone(contactAddressPhoneId)
+
+  @PostMapping(path = ["/contact-address-phone"], produces = [MediaType.APPLICATION_JSON_VALUE])
+  @ResponseBody
+  @Operation(
+    summary = "Creates a new address-specific phone number",
+    description = """
+      Requires role: ROLE_CONTACTS_MIGRATION.
```
[the remark this answers is not in the document]
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/sync/ContactAddressPhoneSyncController.kt:97]: It depends - the actual role is called `ROLE_CONTACTS_MIGRATION` but auth, and the annotations here will accept it is either the long or short version (with or without the ROLE_ prefix).

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/sync/SyncUpdateContactAddressPhoneRequest.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  @Schema(description = "Type of phone", example = "MOB")
+  val phoneType: String,
+
+  @Schema(description = "Phone number", example = "+1234567890")
+  val phoneNumber: String,
+
+  @Schema(description = "Extension number", example = "123")
+  val extNumber: String? = null,
+
+  @Schema(description = "The id of the user who updated the contact phone", example = "JD000001")
```
[the remark this answers is not in the document]
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/sync/SyncUpdateContactAddressPhoneRequest.kt:17]: Will update now. thanks

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/sync/SyncUpdateContactAddressPhoneRequest.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  val updatedBy: String,
+
+  @Schema(description = "The timestamp of when the contact phone was changed", example = "2024-01-01T00:00:00Z")
```
[the remark this answers is not in the document]
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/sync/SyncUpdateContactAddressPhoneRequest.kt:20]: Will clarify, thanks.

**Review by prabash-moj — APPROVED** [review]

**Pull request merged**